File: java/src/main/java/blog/Article.java
package blog;

import io.vavr.collection.List;
import io.vavr.control.Either;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.val;

import java.time.LocalDate;

import static io.vavr.API.Left;
import static io.vavr.API.Right;

@AllArgsConstructor
@Getter
public class Article {
    private final String name;
    private final String content;
    private final List<Comment> comments;

    public Article(String name, String content) {
        this.name = name;
        this.content = content;
        this.comments = List.empty();
    }

    private Either<List<ValidationError>, Article> addComment(
            String text,
            String author,
            LocalDate creationDate) {
        val comment = new Comment(text, author, creationDate);

        return (comments.contains(comment)) ?
                Left(List.of(new ValidationError("Comment already in the article"))) :
                Right(new Article(name, content, comments.append(comment)));
    }

    public Either<List<ValidationError>, Article> addComment(String text, String author) {
        return addComment(text, author, LocalDate.now());
    }
}

File: php/src/Blog/Article.php
<?php

namespace Com\CleanTests\Blog;

class Article
{
    private const DATE_FORMAT = "Ymd";
    private string $name;
    private string $content;
    private array $comments;

    public function __construct(string $name, string $content, array $comments = [])
    {
        $this->name = $name;
        $this->content = $content;
        $this->comments = $comments;
    }

    public function addComment(string $text, string $author): array
    {
        $comment = new Comment($text, $author, date(self::DATE_FORMAT));
        $result = [];

        if (in_array($comment, $this->comments))
            $result[] = new ValidationError("Comment already in the article");
        else $this->comments[] = $comment;

        return $result;
    }

    public function getName(): string
    {
        return $this->name;
    }

    public function getContent(): string
    {
        return $this->content;
    }

    public function getComments(): array
    {
        return $this->comments;
    }
}